crates/configuration/src/shared/node.rs

```diff
@@ -99,23 +100,23 @@ impl NodeConfig {
         Self { args, ..self }
     }
 
-    pub fn being_validator(self) -> Self {
+    pub fn being_validator(self, choice: bool) -> Self {
         Self {
-            is_validator: true,
+            is_validator: choice,
             ..self
         }
     }
 
-    pub fn being_invulnerable(self) -> Self {
+    pub fn being_invulnerable(self, choice: bool) -> Self {
         Self {
-            is_invulnerable: true,
+            is_invulnerable: choice,
             ..self
         }
     }
 
-    pub fn being_bootnode(self) -> Self {
+    pub fn being_bootnode(self, choice: bool) -> Self {
```

Comment:
naming is hard :)

Reply:
Yes, here I chose being_bootnode because with_bootnode didn't make sense and is_bootnode is the name of the getter, if you find better naming, I'm open !

Reply:
I think simply `bootnode` is enough for this..